Integrations/PerceptionPoint/PerceptionPoint.py

```diff
@@ -0,0 +1,177 @@
+try:
+    API_MAX_LOOPS = int(USER_PARAMS.get('api_loops', 1))
+except Exception:
+    API_MAX_LOOPS = 1
+HEADER = {'Authorization': 'Token {}'.format(PP_TOKEN)}
+
+''' CONSTANTS '''
+RELEASE = 'release'
+LIST = 'list'
+API_ACTIONS_DICT = {RELEASE: RELEASE_ENDPOINT,
+                    LIST: INCIDENTS_ENDPOINT}
+SPAM = 'SPM'
+BLOCKED = 'BLK'
+MALICIOUS = 'MAL'
+
+API_CURSOR_ARG = '_cursor'
+
+VERBOSE_VERDICT_PARAM = 'verbose_verdict[]'
+
+FETCH_INCIDENTS_TYPE = [{'demisto_param': 'fetch_malicious',
+                         'req_pname': VERBOSE_VERDICT_PARAM,
+                         'req_pval': MALICIOUS},
+                        {'demisto_param': 'fetch_blocked',
+                         'req_pname': VERBOSE_VERDICT_PARAM,
+                         'req_pval': BLOCKED},
+                        {'demisto_param': 'fetch_spam',
+                         'req_pname': VERBOSE_VERDICT_PARAM,
+                         'req_pval': SPAM}]
+
+''' HELPER FUNCTIONS '''
+
+
+def build_fetch_incident_types(fetch_select):
+    fetch_type_dict = defaultdict(list)  # type: ignore
+    for darg in FETCH_INCIDENTS_TYPE:
+        darg_input = fetch_select.get(darg['demisto_param'])
+        if darg_input:
+            fetch_type_dict[darg['req_pname']].append(darg.get('req_pval', darg_input))
+    return dict(fetch_type_dict)
+
+
+def create_incident(record):
+    record.pop('Attachment', None)
+    record['RawJSON'] = json.dumps(record)
+    return record
+
+
+def collect_incidents(**kwargs):
+    list_url = build_request_url(LIST)
+    api_res = get_pp_api_result(url=list_url, **kwargs)
+    num_of_results = api_res.get('count')
+    incidents = []  # type: list
+    api_loops = 0
+    while num_of_results and api_loops < API_MAX_LOOPS:
+        incidents += map(create_incident, api_res.get('results'))
+        if api_res.get('next'):
+            api_res = get_pp_api_result(url=api_res.get('next'))
+            num_of_results = api_res.get('count')
+        api_loops += 1
+    return incidents
+
+
+def report_incidents(incidents_list):
+    demisto.incidents(incidents_list)
+
+
+def get_pp_api_result(url, **kwargs):
+    try:
+        res = requests.get(url=url,
+                           params=kwargs,
+                           headers=HEADER,
+                           verify=SECURED)
+        res.raise_for_status()
+        try:
+            res_content = res.json()
+        except Exception:
+            res_content = {}
+        return res_content
+    except requests.exceptions.HTTPError as err:
+        if 400 <= res.status_code < 500:
+            return_error('Invalid token')
+        else:
+            return_error(err)
+    except Exception as err:
+        return_error(err)
+
+
+def build_request_url(api_action):
+    return URL.format(endpoint=API_ACTIONS_DICT.get(api_action))
+
+
+def command_fetch_incidents():
+    try:
+        args_dict = demisto.args()
```

Comment:
please do not cascade arguments with kvargs.
extract demisto args explicitly in the beginning of _command function and pass them to the business object function explicitly.